### PR TITLE
ScalaDocs improved with packaging, class, object, trait, and method l…

### PR DESCRIPTION
**Pull request by riverma**
- new package-level documentation to direct ScalaDocs viewers on role of each package
- improved class, object, and trait level documentation to make ScalaDocs more useful and complete

### CODE REVIEW

File: src/main/scala/org/dia/loaders/NetCDFReader.scala [hunk 1]
```diff
@@ -57,6 +57,9 @@ object NetCDFReader {
 
   /**
    * Gets just the variable names of a NetCDF at some URI.
+   *
+   * @param uri
```
Comment (chrismattmann) [inline src/main/scala/org/dia/loaders/NetCDFReader.scala]: can we describe the params too?

File: src/main/scala/org/dia/loaders/MergReader.scala [hunk 2]
```diff
@@ -54,6 +79,14 @@ object MergReader {
     SourceArray
   }
 
+  /**
+   * Returns a Java array loaded from a MERG byte array at a specified offset based on a specified shape
+   *
+   * @param byteArray
+   * @param offset
```
Comment (chrismattmann) [inline src/main/scala/org/dia/loaders/MergReader.scala]: same comments here, define the params too

File: src/main/scala/org/dia/loaders/NetCDFReader.scala [hunk 3]
```diff
@@ -70,6 +73,12 @@ object NetCDFReader {
     list
   }
 
+  /**
+   *
+   * @param dataset
```
Comment (chrismattmann) [inline src/main/scala/org/dia/loaders/NetCDFReader.scala]: define params

File: src/main/scala/org/dia/loaders/NetCDFReader.scala [hunk 4]
```diff
@@ -86,6 +95,12 @@ object NetCDFReader {
     coordinateArray
   }
 
+  /**
+   *
+   * @param name
```
Comment (chrismattmann) [inline src/main/scala/org/dia/loaders/NetCDFReader.scala]: define params

File: src/main/scala/org/dia/loaders/MergReader.scala [hunk 5]
```diff
@@ -35,16 +35,41 @@ import scala.io.Source
  */
 object MergReader {
 
+  /**
+   * Returns a Java 1-dimensional array loaded from a shape array at a specified offset
+   *
+   * @param shape
+   * @param offset
```
Comment (chrismattmann) [inline src/main/scala/org/dia/loaders/MergReader.scala]: define params

**Comment by chrismattmann** [comment]
as a first cut this is great @riverma note my comments about adding in docs for the params too. I think once we do that this can be merged.

**Comment by riverma** [comment]
Hi @chrismattmann - thanks for the comments. Your requested modifications were added to a second commit in the scaladocs branch. Should be good to go, but let me know if I missed something.

**Comment by chrismattmann** [comment]
great work @riverma thanks!

**Pull request merged**